src/query/functions/src/scalars/map.rs

```diff
@@ -244,4 +246,50 @@ pub fn register(registry: &mut FunctionRegistry) {
                     .any(|(k, _)| k == key)
             },
         );
+
+    registry.register_3_arg_core::<EmptyMapType, GenericType<0>, GenericType<1>, MapType<GenericType<0>, GenericType<1>>, _, _>(
+        "map_insert",
+        |_, _, _, _| FunctionDomain::Full,
+        |_, key, value, ctx| {
+            let mut b = ArrayType::create_builder(1, ctx.generics);
```

Comment:
The data type of key can only be `bool`, `string`, `number`, `date` and `timestamp`, we need check the key type first. Here is an example in `map` function:
https://github.com/datafuselabs/databend/blob/main/src/query/functions/src/scalars/map.rs#L52